letsencrypt/plugins/webroot.py

```diff
@@ -98,6 +99,10 @@ def prepare(self):  # pylint: disable=missing-docstring
                          self.full_roots[name])
             try:
                 os.makedirs(self.full_roots[name])
+                # Set permissions as parent directory (GH #1389)
+                filemode = stat.S_IMODE(os.stat(path).st_mode)
+                os.chmod(self.full_roots[name], filemode)
```

Comment:
> Question: is the reason to call `os.makedirs(file)` then `os.chmod(file,
> filemode)` rather than just `os.makedirs(file, filemode)` because

According to https://stackoverflow.com/questions/5231901/permission-problems-when-creating-a-dir-with-os-makedirs-python os.makedirs's mode parameter is 
not always honored. If that's OK I can add a comment with a link to that.